src/js/components/PlacementConstraintsFrameworkAdapter.js

```diff
@@ -0,0 +1,102 @@
+import React, { Component } from "react";
+import Batch from "#SRC/js/structs/Batch";
+import Transaction from "#SRC/js/structs/Transaction";
+import { deepCopy } from "#SRC/js/utils/Util";
+import TransactionTypes from "#SRC/js/constants/TransactionTypes";
+import PlacementConstraintsPartial
+  from "#SRC/js/components/PlacementConstraintsPartial";
+import JSONSingleContainerParser
+  from "#PLUGINS/services/src/js/reducers/JSONSingleContainerParser";
+import CreateServiceModalFormUtil
+  from "#PLUGINS/services/src/js/utils/CreateServiceModalFormUtil";
+
+export default class PlacementConstriantsFrameworkAdapter extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      batch: new Batch(),
+      appConfig: this.getInitialAppConfig()
+    };
+  }
+
+  getInitialAppConfig() {
+    // parse
+    const arrayified = JSON.parse(this.props.data);
+
+    // add a transaction for each in the array of given constraints
+    const batch = new Batch();
+
+    return this.getAppConfig(batch);
+  }
+
+  getAppConfig(batch = this.state.batch, baseConfig = {}) {
+    const baseConfigCopy = deepCopy(baseConfig);
+    const newConfig = batch.reduce(JSONSingleContainerParser, baseConfigCopy);
+
+    return CreateServiceModalFormUtil.stripEmptyProperties(newConfig);
+  }
+
+  // direct copy paste
```

Comment:
We might want to have a higher order component for the `Batch`. That would implement all the `onChange` mechanics. So that we could wrap the Partial with the component and avoid copy-pasting. What do you think @Poltergeist ?

Reply:
Yes  that would be very helpful, I think something like `<BatchForm>`